client/src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,56 @@
+import { Container, Grid } from '@material-ui/core';
+import React from 'react';
+import useStyles from './useStyles';
+import KanbanLogo from '../../Images/logo.png';
+import { DashboardOutlined, CalendarTodayOutlined } from '@material-ui/icons';
+import NavButton from '../Button/NavButton';
+import CreateBoardButton from '../Button/CreateBoardButton';
+import AvatarButton from '../AvatarButton/AvatarButton';
+import AvatarMenu from '../AvatarMenu/AvatarMenu';
+
+export default function Navbar(): JSX.Element {
+  const classses = useStyles();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <Container className={classses.root}>
+      <Grid container spacing={2}>
+        <Grid item md={3} className={classses.container}>
+          <img src={KanbanLogo} alt="kanban-logo" />
+        </Grid>
+        <Grid item md={6} className={classses.navButtonContainer}>
+          <NavButton title="Dashboard" to="/" icon={<DashboardOutlined />} />
+          <NavButton title="Calender" to="/calender" icon={<CalendarTodayOutlined />} />
+        </Grid>
+        <Grid item md={3} className={classses.avatarContainer}>
+          <CreateBoardButton
+            onClick={() => {
+              // handles the create board functionality
+            }}
+          />
+          <AvatarButton onClick={handleClick} />
+          <AvatarMenu
+            id="menu"
+            anchorEl={anchorEl}
+            open={Boolean(anchorEl)}
+            onClose={handleClose}
+            onClickProfile={() => {
+              // handles profile option in the menu
+            }}
+            onClickLogOut={() => {
+              // handles logout option in the menu
```

Comment:
We should have this functionality available in the context, we could re-use that in the immer state work?

Reply:
Yes your correct, we would have a function for logout that updates the state when the user logs out.